ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py

```diff
@@ -12,6 +10,48 @@
 from ivy_tests.test_ivy.helpers import handle_frontend_test
 
 
+# nanmedian
+@handle_frontend_test(
+    fn_tree="numpy.nanmedian",
+    dtype_and_a=statistical_dtype_values(function="nanmedian"),
+    dtype=helpers.get_dtypes("float", full=False, none=True),
+    where=np_frontend_helpers.where(),
+    keep_dims=st.booleans(),
+)
+def test_numpy_nanmedian(
+    dtype_and_a,
+    dtype,
+    where,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+    keep_dims,
+):
+    input_dtypes, a, axis = dtype_and_a
+    if isinstance(axis, tuple):
+        axis = axis[0]
+    where, input_dtypes, test_flags = np_frontend_helpers.handle_where_and_array_bools(
+        where=where,
+        input_dtype=input_dtypes,
+        test_flags=test_flags,
+    )
+    np_frontend_helpers.test_frontend_function(
+        input_dtypes=input_dtypes,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        a=a[0],
+        axis=axis,
+        dtype=dtype[0],
```

Comment:
E   TypeError: nanmedian() got an unexpected keyword argument 'dtype'

Reply:
please do not pass in not existing arguments for the test

Reply:
I do not understand the error I made my test function the same as other functions in the same file (test_averages_and_variances) I did like the mean and nanmean functions. these functions are merged. I do not know where is the problem in my finction and there is not clear documentation guide me to know what to do.
can you help me with more details?

Reply:
you need to understand that the test need to be customly made for each function as the behavior of those functions are not the same. You can read the contribution guide and see that you need to generate inputs for your function by writing down generation strageties like "dtype_and_a=statistical_dtype_values(function="nanmedian")" and then pass the generated value in to the np_frontend_helpers.test_frontend_function along with some other environment configration variables like frontend, test_flags, fn_tree and on_device. So you do not need to generate and pass in that "dtype" argument.

Reply:
you need to generate inputs for all of the argments(a, axis, keepdims, overwrite_input) of your function while "out" will be handled automatically